src/react-components/src/__tests__/components/UnorderedList-test.tsx

```diff
@@ -0,0 +1,35 @@
+import { expect } from "chai";
+
+import * as Enzyme from "enzyme";
+import * as React from "react";
+
+import UnorderedList from "../../components/List/UnorderedList";
+
+describe("Link With Underline", () => {
+    let wrapper: Enzyme.ReactWrapper<{}, {}>;
+    let textList: JSX.Element[] = [];
+
+    it("Returns proper classes for UnorderedList with modifier", () => {
```

Comment:
These tests are really testing bem, so we probably don't need it here.  

If there aren't UnorderedList tests, I think we want to test on mounting the `<UnorderedList>` and seeing `<ul>`s and `<li>`s

Reply:
@hokei I added these tests because the 🐛 bug, in this case, was due to the values being passed into BEM in a different order than expected. These tests are to tests that the unorderedList is sending the values from its props to the BEM function in the appropriate order.

Reply:
I agree with @hokei on this one—this test should live elsewhere since all components could run into this issue.

Reply:
I want to test that the bem function is called with the parameters in the proper order like:    
`unorderedList` sends the values in the right order like: 
```
<div className={bem(baseClass, modifiers, "list-container")}>
         {/**/}
        </div>
```
instead of `bem("list-container", modifiers, baseClass)`. This test is specific to how bem() is called within the OrderedList and without this test we could run into a similar bug in this same component if for some reason the order for how params in unorderedList are passed into the bem function changes

Reply:
Do you all think that bem should accept an object with named parameters instead of just passing in 3 values so it would accept something like

`bem({baseClass, modifiers, blockName: "list-container }` instead of `bem(baseClass, modifiers, "list-container")`? Without this change I think we could run into issues in other components passing the parameters to the bem function in a different order than what is expected.

Reply:
So to make sure I'm understanding this correctly, would we do something like this every time we called the bem util?

`bem({baseClass: unordered_list_baseclass, modifiers: unordered_list_modifiers_array, blockName: "list-container"})`

Reply:
That's what I was proposing unless you have other recommendations or think it's too verbose but I think it could reduce the chance of swapping values when they are being passed into the bem function.

Reply:
I think this is a good test - though if we're using it in many places I'd love to see it extracted into a utility, something like (checkBem?) so that all the tests have access to it (doesn't have to happen here, only if we're starting to use it everywhere). 

In general, I'd like stuff like this to be tested with visual regression testing, rather than specific class names, which are frustrating to change if there are lots of tests that are looking for specific class names.  (We don't really care that the `unordered-list unordered-list unordered-list--scroll` classes are called, we care that there is a scrollbar).  But setting that up isn't even on the roadmap at this point, so I think this is the closest we can get.

Reply:
I actually don't oppose _a_ test like the one we have here simply because it helps with understanding BEM imo (Crystal converted the original BEM util from another repo here at NYPL and for a while we were both pretty confused at what the output should be vs. what it was).

But I agree, visual regression testing is something to look for in the future. We can talk about getting it in the backlog at next planning.